module.nix

```diff
@@ -177,7 +178,12 @@ in
             host = mkDefault "";
           };
           cert_discovery_dir = mkIf (cfg.nginx.enable && cfg.nginx.enableACME) "env://CREDENTIALS_DIRECTORY";
-          paths.media = mkDefault "/var/lib/authentik/media";
+          storage.media = {
+            backend = mkDefault "file";
+            file = mkDefault {
```

Comment:
Could you add a `mkRenamedOptionModule` import the catches the old setting?

Reply:
Does this even work for `freeformType`? Since this is not part the exposed API of the module but the config of authentik itself, if a user would override this, it's their duty to keep up with changes of the config format.

Reply:
With `attrsOf submodule` etc. it would definitely not work, but I don't recall how it was for just submodules.
Will check and report back.

Reply:
OK yeah it checks if the attr path exists in `options` which is not the case for (freeform) submodules.

I think it's also common in nixpkgs to not write assertions for dropped freeform attributes. Though it would certainly be doable using `assertions` in the top-level scope or `apply`.
@WilliButz I don't have a strong opinion, so your call I'd say :)